andela/rivendell-ah
# feat(reset): implement reset password

#### What does this PR do?

- Users should be able to reset password reset via email

#### Description of Task to be completed?
- Password reset functionality is missing in our application. Its invaluable in instances where our users forget their login credentials.

#### How should this be manually tested?

- Run `npm install`
- Run `sequelize db:migrate` to create the table on the database
- Use postman to test the route POST `/api/v1/users/forgot-password`
ensure to supply the following fields:
```
{
"user": {
"email": "example@yahoo.com",
}
}
```
- User should check his/her mail for the password reset link
- User should click the link so that he/she can have access to the token from the url,
- User can now reset his/her password in postman PUT `/api/users/reset-password`
```
`{
"user": {
"password": "123456s",
"confirm": "123456s"
}
}

set header: `Authorization: token `
```

- Run `npm test`
#1159204695

#### Any background context you want to provide?
N/A

#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-08-03 at 3 01 53 pm" src="https://user-images.githubusercontent.com/38965909/43646991-48cd7682-972e-11e8-921a-0c1550d9bd76.png">
<img width="1440" alt="screen shot 2018-08-03 at 3 03 16 pm" src="https://user-images.githubusercontent.com/38965909/43647046-69a497be-972e-11e8-914f-15ef14b13dd1.png">
<img width="1440" alt="screen shot 2018-08-03 at 3 05 16 pm" src="https://user-images.githubusercontent.com/38965909/43647208-df32976a-972e-11e8-8a63-c82fcff15ffd.png">
